packages/mobile/src/i18n/index.ts

```diff
@@ -13,13 +13,34 @@ import { getOtaTranslations } from 'src/i18n/otaTranslations'
 
 const TOS_LINK_DISPLAY = TOS_LINK.replace(/^https?:\/\//i, '')
 
-async function getAvailableResources(cachedTranslations: Resource) {
+// used to prevent translations for all languages from being loaded upfront
+const translationResource: Resource = {}
+
+const mergeTranslationResources = (
+  res1: ResourceLanguage,
+  res2: ResourceLanguage,
+  language: string
+) => {
+  if (!translationResource[language]) {
+    translationResource[language] = _.merge(res1, res2)
+  }
+
+  return translationResource[language]
+}
```

Comment:
What do you think of putting this logic directly in the for loop?
This way we can replace the global object with a local variable captured by the getter.

Main reason is to avoid a global var for this.